src/crawlee/http_clients/_base.py

```diff
@@ -87,12 +93,38 @@ async def send_request(
 
         Raises:
             ProxyError: Raised if a proxy-related error occurs.
+            HttpStatusError: Raised if the response status code indicates an error.
 
         Returns:
             The HTTP response received from the server.
         """
 
+    def _raise_for_error_status_code(
+        self,
+        status_code: int,
+        additional_http_error_status_codes: set[int],
+        ignore_http_error_status_codes: set[int],
+    ) -> None:
+        """Raise an exception if the given status code is considered as an error."""
+        exclude_error = status_code in ignore_http_error_status_codes
+        include_error = status_code in additional_http_error_status_codes
+
+        if include_error or (self._is_status_code_error(status_code) and not exclude_error):
+            if include_error:
+                raise HttpStatusCodeError(f'Status code {status_code} (user-configured to be an error) returned.')
+
+            raise HttpStatusCodeError(f'Status code {status_code} returned.')
+
+    def _is_status_code_error(self, value: int) -> bool:
+        """Returns `True` for 4xx or 5xx status codes, `False` otherwise."""
+        return self._is_status_code_client_error(value) or self._is_status_code_server_error(value)
+
+    @staticmethod
+    def _is_status_code_client_error(value: int) -> bool:
+        """Returns `True` for 4xx status codes, `False` otherwise."""
+        return 400 <= value <= 499  # noqa: PLR2004
+
     @staticmethod
-    def _is_server_code(status_code: int) -> bool:
-        """Helper to determine if a status code is a server error."""
-        return 500 <= status_code <= 599  # noqa: PLR2004
+    def _is_status_code_server_error(value: int) -> bool:
+        """Returns `True` for 5xx status codes, `False` otherwise."""
+        return 500 <= value <= 599  # noqa: PLR2004
```

Comment:
how do we handle error codes above 599 btw?

Reply:
We do not, yet - how do we want to approach them?

Reply:
all error codes above 500 should be handled the same way